src/codeu/chat/util/dbUtils.java

```
@@ -0,0 +1,152 @@
+package codeu.chat.util;
+
+import java.sql.*;
+
+import java.util.Comparator;
+import codeu.chat.common.User;
+import codeu.chat.common.Conversation;
+import codeu.chat.common.Message;
+import codeu.chat.util.store.Store;
+
+public class dbUtils {
+
+    private final static Logger.Log logger = Logger.newLog(dbUtils.class);
+
+    private static final Comparator<Uuid> UUID_COMPARE = new Comparator<Uuid>() {
+
+        @Override
+        public int compare(Uuid a, Uuid b) {
+
+            if (a == b) {
+                return 0;
+            }
+
+            if (a == null && b != null) {
+                return -1;
+            }
+
+            if (a != null && b == null) {
+                return 1;
+            }
+
+            final int order = Integer.compare(a.id(), b.id());
+            return order == 0 ? compare(a.root(), b.root()) : order;
+        }
+    };
+
+    private Connection connect() {
+        // SQLite connection string
+        String url = "jdbc:sqlite:database/main.db";
+        Connection conn = null;
+        try {
+            conn = DriverManager.getConnection(url);
+        } catch (SQLException e) {
+            logger.error("Connection to database failed.", e.getMessage());
```

This swallows an exception that could lead to the caller getting a null connection. Perhaps we should create our own database error class and raise that when there are problems?